### PR TITLE
Get envoy TLS cert from AKV

### PR DESCRIPTION
**Pull request by takuro-sato**
Get envoy TSL cert and key from Azure Key Vault following the GCP implementation.
Introduce AZURE_BA_PARAM_KEY_VAULT_URL, AZURE_BA_METADATA_OPERATOR, and AZURE_BA_METADATA_ENVIRONMENT.

GCP implementation (with Azure terminology) is:
- Fetch subscription ID from metadata service (local server to provide metadata) to specify AKV's instance.
- Fetch 'operator' and 'environment' value from metadata service. (metadata service has key value store feature)
- They are used to specify key name and cert name

We concluded we can't do the exact same because ACI doesn't have metadata service that meet the requirement.
Now we won't use ACI, but for now we stick to environment variable instead of metadata servicer-ish solution, because environment variable is widely available and would help configuration for local development.

### CODE REVIEW

File: third_party/container_deps.bzl [hunk 1]
```diff
@@ -35,7 +35,7 @@ def container_deps():
         "envoy-distroless": {
             "arch_hashes": {
                 # v1.23.1
-                "amd64": "e2c642bc6949cb3053810ca14524324d7daf884a0046d7173e46e2b003144f1d", # @envoy-distroless-amd64-repo-digests-replace-marker@
+                "amd64": "e2c642bc6949cb3053810ca14524324d7daf884a0046d7173e46e2b003144f1d",  # @envoy-distroless-amd64-repo-digests-replace-marker@
```
Comment (takuro-sato) [inline third_party/container_deps.bzl:38]: edited by the pre-commit

File: production/packaging/azure/seller_frontend_service/bin/init_server_basic [hunk 2]
```diff
@@ -23,10 +23,24 @@ set -x
...
+
+# Get credentials (to be used for fetching secrets)
+ACCESS_TOKEN=$(wget -q --header "Metadata:true"  --output-document -  \
+"${AZURE_BA_PARAM_GET_TOKEN_URL}?api-version=2018-02-01&resource=https%3A%2F%2Fvault.azure.net%2F" \
+| /server/bin/jq -r '.access_token')
+
+# Get our secrets, parse, decode, and write the secret strings to files that the Envoy config will read.
+## TLS Key
+wget -q --header "Authorization: Bearer ${ACCESS_TOKEN}" --output-document - \
+${AZURE_BA_PARAM_KEY_VAULT_URL}/secrets/${AZURE_BA_METADATA_OPERATOR}-${AZURE_BA_METADATA_ENVIRONMENT}-SfeTlsKey?api-version=7.4 \
```
Comment (beejones) [inline production/packaging/azure/seller_frontend_service/bin/init_server_basic:35]: do we have a process to roll this KV secret when the cert expires?

Comment (takuro-sato) [reply production/packaging/azure/seller_frontend_service/bin/init_server_basic:35]: No. That's something we need to discuss in future.

What I know about it so far is:
- Looks like there is no code to rotate these cert
- The cert is loaded start up time, so the rotation requires restart of the service instance

Comment (beejones) [reply production/packaging/azure/seller_frontend_service/bin/init_server_basic:35]: Can  you create an issue, so we keep track of this?

Comment (takuro-sato) [reply production/packaging/azure/seller_frontend_service/bin/init_server_basic:35]: Sure https://msr-gryffindor.visualstudio.com/Protected-Audiences/_workitems/edit/1341

**Review by beejones — APPROVED** [review]

**Pull request merged**